core/base-service/coalesce-badge.js

```diff
@@ -93,12 +93,6 @@ export default function coalesceBadge(
   } = defaultBadgeData
 
   let style = coalesce(overrideStyle, serviceStyle)
-  if (typeof style !== 'string') {
-    style = 'flat'
-  }
```

Comment:
Without this, `startsWith` below would blow up for numbers and other types. `includes` below will be much more forgiving when such values are provided. I believe we can get rid of this check as well.